File: 6. Python/EX41.py
# Inicializa variáveis
i = 0
tot = 0
qtdRat = 0
qtdSap = 0
qtdCoe = 0

# Solicita número de casos de teste
n = int(input("# Quantos casos de teste serao digitados: "))

# Verifica se o número de casos é válido
if n > 0:
    while i < n:
        # Solicita quantidade e tipo de cobaia
        qtd = int(input("\n# Quantidade de cobaias: "))
        tip = input("# Tipo de cobaia: ")[0].upper()  # Captura a primeira letra

        # Atualiza as quantidades com base no tipo de cobaia
        switch = {
            'R': lambda: globals().__setitem__('qtdRat', globals()['qtdRat'] + qtd),
            'S': lambda: globals().__setitem__('qtdSap', globals()['qtdSap'] + qtd),
            'C': lambda: globals().__setitem__('qtdCoe', globals()['qtdCoe'] + qtd)
        }

        # Verifica e executa a função correta
        if tip in switch:
            switch[tip]()
            i += 1
        else:
            print("# Cobaia invalida !")
            continue

    tot = qtdRat + qtdSap + qtdCoe  # Calcula o total de cobaias

else:
    print("# Numeros de casos invalidos !")

# Verifica se há cobaias registradas
if tot > 0:
    # Exibe o relatório final
    print("\n*** RELATORIO FINAL ***")
    print(f"# Total: {tot}")
    print(f"# Total de coelhos: {qtdCoe}")
    print(f"# Total de ratos: {qtdRat}")
    print(f"# Total de sapos: {qtdSap}")
    print(f"# Percentual de coelhos: {((qtdCoe / tot) * 100):.2f}")
    print(f"# Percentual de ratos: {((qtdRat / tot) * 100):.2f}")
    print(f"# Percentual de sapos: {((qtdSap / tot) * 100):.2f}")
else:
    print("# Quantidade invalida !")

File: 4. Java/EX41.java
package Apostila;

import java.util.Locale;
import java.util.Scanner;

public class EX41
{
    public static void main(String[] args)
    {
        // Variáveis para tipo e contagem de cobaias
        char tip;
        int i = 0, n, qtd, tot = 0;
        int qtdRat = 0, qtdSap = 0, qtdCoe = 0;
            
        // Define o locale e inicializa o Scanner
        Locale.setDefault(Locale.US);
        
        // Solicita o número de casos
        try (Scanner ent = new Scanner(System.in))
        {
            // Solicita o número de casos
            System.out.print("# Quantos casos de teste serao digitados: ");
            n = ent.nextInt();
            
            // Processa casos positivos
            if (n > 0)
            {
                while (i < n)
                {
                    // Solicita quantidade de cobaias
                    System.out.print("\n# Quantidade de cobaias: ");
                    qtd = ent.nextInt();
                    
                    // Solicita tipo de cobaia
                    System.out.print("# Tipo de cobaia: ");
                    tip = ent.next().toUpperCase().charAt(0);
                    
                    // Acumula a quantidade de cada tipo usando switch
                    switch (tip) 
                    {
                        case 'R':
                            qtdRat += qtd; // Acumula ratos
                            break;
                            
                        case 'S':
                            qtdSap += qtd; // Acumula sapos
                            break;
                        
                        case 'C':
                            qtdCoe += qtd; // Acumula coelhos
                            break;
                        
                        default:
                            System.out.println("# Opcao invalida !"); // Opção inválida
                            i--; // Não conta este caso
                    }

                    i++;   
                }       
                
                // Total de cobaias
                tot = qtdRat + qtdSap + qtdCoe;
            }
            
            // Caso inválido
            else
            {
                System.out.println("# Numeros de casos invalidos !");
            }
            
            // Relatório final
            if (tot > 0)
            {
                System.out.println("\n*** RELATORIO FINAL ***");
                System.out.printf("# Total: %d\n", tot);
                System.out.printf("# Total de coelhos: %d\n", qtdCoe);
                System.out.printf("# Total de ratos: %d\n", qtdRat);
                System.out.printf("# Total de sapos: %d\n", qtdSap);
                System.out.printf("# Percentual de coelhos: %.2f %%\n", (((double) qtdCoe / tot) * 100));
                System.out.printf("# Percentual de ratos: %.2f %%\n", (((double) qtdRat / tot) * 100));
                System.out.printf("# Percentual de sapos: %.2f %%\n", (((double) qtdSap / tot) * 100));
            }
        }
    }
}
